nncf/experimental/tensorflow/quantization/init_range.py

```diff
@@ -0,0 +1,116 @@
+from nncf.experimental.tensorflow.nncf_network import NNCFNetwork
+from nncf.tensorflow.tensor_statistics.statistics import tf_convert_stat_to_min_max_tensor_stat
+
+
+class TFRangeInitParamsV2(TFRangeInitParams):
+    def get_init_config_for_quantization_point_v2(self,
+                                                  node_name: str,
+                                                  input_type: str) -> RangeInitConfig:
+        group = QuantizerGroup.WEIGHTS if input_type == InputType.WEIGHTS else QuantizerGroup.ACTIVATIONS
+        return self.get_init_config_for_scope_and_group(node_name, group)
+
+
+def _get_reduction_shape(tensor_shape, channel_axes, per_channel):
+    ndims = len(tensor_shape)
+    channel_axes_ = channel_axes if isinstance(channel_axes, (list, tuple)) else [channel_axes]
+    reduction_shape = get_axes(ndims, per_channel, channel_axes_)
+    return tuple(reduction_shape)
+
+
+class RangeInitializerV2(RangeInitializer):
+    def __init__(self, range_init_params: TFRangeInitParamsV2):
+        super().__init__(range_init_params)
+        self.nncf_quantization_operation_classes = NNCF_QUANTIZATION_OPERATIONS_V2.registry_dict.values()
+
+    def _register_op_collector(self, op, collectors, handles):
+        node_name = ''  # TODO(andrey-churkin): Use correct node_name
+        init_config = self.range_init_params.get_init_config_for_quantization_point_v2(
+            node_name,
+            op.input_type
+        )
+
+        is_weights = op.input_type == InputType.WEIGHTS
+        collector_params = RangeInitCollectorParams(is_weights, op.mode, op.per_channel)
+
+        reduction_shape = _get_reduction_shape(op.input_shape, op.channel_axes, op.per_channel)
+        if is_weights:
+            num_batches = 1
+        else:
+            num_batches = int(np.ceil(init_config.num_init_samples / self.dataset.batch_size))
+
+            per_sample_stats = init_config.init_type in ['mixed_min_max', 'mean_min_max']
+            if collector_params.use_per_sample_stats(per_sample_stats):
+                reduction_shape = reduction_shape[1:]
+
+        collector = RangeInitializerV2.generate_stat_collector(
+            reduction_shape,
+            collector_params,
+            init_config,
+            num_batches
+        )
+        handles.append(op.register_hook_pre_call(collector.register_input))
+        op.enabled = False
+        collectors.append((op, collector))
+
+    def run(self, model: NNCFNetwork) -> None:
+        handles = []
+        collectors = []
+        for op in model.nncf_operations:
+            if op.__class__ not in self.nncf_quantization_operation_classes:
+                continue
+            self._register_op_collector(op, collectors, handles)
+
+        for (x, _) in ProgressBar(
+                islice(self.dataset, self.num_steps),
+                total=self.num_steps,
+                desc='Collecting tensor statistics/data'
+        ):
+            model(x, training=False)
```

Comment:
In PT, we have `TensorStatisticCollectionBuilder` and `TensorStatisticCollectionController`, to separate the responsibilities of statistic collection and range initialization. I recommend doing the same for TF.

Reply:
Great comment! I suggest creating a separate ticket for these improvements. In the current PR, I took the `init_range` module from the current develop branch and added a minimal number of changes to get the working example.
@vshampor @alexsu52 Do you have any concerns about these?

Do these improvements affect only the experimental version or version from the develop branch too?

Reply:
I created the ticket (Ref: 79913) for this task.